tsinfer/inference.py

```diff
@@ -490,6 +498,136 @@ def match_samples(
     return ts
 
 
+def match_ancestors_including_noncontemporanous_samples(
```

Comment:
An obvious question here is, do we really need another function for this? Could we do the same thing as this via ``match_ancestors``? Or, more to the point, why aren't we doing this as a method of the AncestorData file, since the majority of what we're doing is just splicing in some extra haplotypes to create a new ancestor data file.

Reply:
I was thinking we could either (a) roll this into the match_ancestors function, or as you suggest (b) have an ancestor data merge function. The slight catch with (b) is that if we are using "true" sample ancestors, then we perhaps want to think of the thing in the ancestor_data file as a true genome, rather than an inferred ancestor - for example, we want an individual associated with the ancestor (we don't have a concept of individuals in the ancestor_data file), and we probably don't want the "ancestor_data_id=X" metadata field.